bin/provisioner.js

```diff
@@ -1 +1,95 @@
-//TODO: The same as server but without configuring api...
\ No newline at end of file
+'use strict';
+
+var debug = require('debug')('aws-provisioner:bin:provisioner');
+var Promise = require('promise');
+var base = require('taskcluster-base');
+var provision = require('../provisioner/provision');
+var aws = require('multi-region-promised-aws');
+var data = require('../provisioner/data');
+var AwsManager = require('../provisioner/aws-manager');
+var awsPricing = require('../provisioner/aws-pricing');
+var Cache = require('../lib/cache');
+var taskcluster = require('taskcluster-client');
+
+
+
+var launch = function(profile) {
+  var cfg = base.config({
+    defaults: require('../config/defaults.js'),
+    profile: require('../config/' + profile),
+    filename: 'taskcluster-aws-provisioner',
+    envs: [
+        'provisioner_publishMetaData',
+        'provisioner_awsInstancePubkey',
+        'provisioner_awsKeyPrefix',
+        'taskcluster_queueBaseUrl',
+        'taskcluster_authBaseUrl',
+        'taskcluster_credentials_clientId',
+        'taskcluster_credentials_accessToken',
+        'pulse_username',
+        'pulse_password',
+        'aws_accessKeyId',
+        'aws_secretAccessKey',
+        'azure_accountName',
+        'azure_accountKey',
+        'influx_connectionString'
+    ]
+  });
+
+  var allowedRegions = cfg.get('provisioner:allowedRegions').split(',');
+  var keyPrefix = cfg.get('provisioner:awsKeyPrefix');
+  var pubKey = cfg.get('provisioner:awsInstancePubkey');
+  var provisionerId = cfg.get('provisioner:id');
+
+  var influx = new base.stats.Influx({
+    connectionString:   cfg.get('influx:connectionString'),
+    maxDelay:           cfg.get('influx:maxDelay'),
+    maxPendingPoints:   cfg.get('influx:maxPendingPoints')
+  });
+
+  var WorkerType = data.WorkerType.setup({
+    table: cfg.get('provisioner:workerTypeTableName'),
+    credentials: cfg.get('azure'),
```

Comment:
Oh, pretty sure I've said this before... But let's use auth.tc.net for delegating access to the azure account.
It's much easier as you don't have put azure credentials in env vars... just the tc creds, see:
https://github.com/taskcluster/taskcluster-index/blob/master/bin/server.js#L38-L41

In this case:
```js
  var WorkerType = data.WorkerType.setup({
    account: cfg.get('provisioner:azureAccount').
    table: cfg.get('provisioner:workerTypeTableName'),
    credentials: cfg.get('taskcluster'),
    authBaseUrl:      cfg.get('taskcluster:authBaseUrl'),
    context: {
      keyPrefix: keyPrefix,
      provisionerId: provisionerId,
    },
  });
```
`authBaseUrl` is not strictly required, but it's nice... If you do this have you have sufficient scopes you just configure `provisioner:azureAccount` to `taskclusterdev` when testing... and to `taskcluster` in production and you're good to go...

Note, you need to change this in `server.js` too... fix configs and env vars loaded...